dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/ResultsParser.java

```diff
@@ -1,17 +1,169 @@
+import java.io.FileNotFoundException;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Scanner;
 
 public class ResultsParser {
 
+    // name of line to get values from in Parallelization technique
+    public static final String PARALLELIZATION = "New order time:";
+    // name of line to get values from in selection and prioritization
+    public static final String APFD_VALUE = "APFD value:";
+
+    /*
+     * A private method to search a file for a keyword and return the value that follows
+     * that keyword
+     *
+     * @return the data value without any leading or trailing whitespaces
+     */
+    private static String parseFile(File file, String keyword) throws FileNotFoundException {
+
+        Scanner scanner = new Scanner(file);
+        while (scanner.hasNextLine()) {
+            String currLine = scanner.nextLine();
+            // lineNum++;
+            if (currLine.contains(keyword)) {
+                // gets numeric value of data
+                String data = currLine.substring(keyword.length(), currLine.length());
+                scanner.close(); // close Scanner before returning
+                // trim away any whitespaces leading or after the data value
+                return data.trim();
+
+            }
+        }
+        scanner.close();
+        return null; // none of the lines contained the keyword
+    }
+
+    /*
+     * A private method that returns the name of the argument specified by the flag
+     *
+     * @return the name of the argument corresponding to flag
+     */
+    private static String getArgName(List<String> argsList, String flag) {
+        String flagName = null;
+        int index = argsList.indexOf(flag);
+        if (index != -1) {
+            int nameIndex = index + 1;
+            if (nameIndex >= argsList.size()) {
+                System.err.println(flag + " argument is specified but flagName is not." + " Please use the format: "
+                        + flag + " flagName");
+                System.exit(0);
+            }
+            flagName = argsList.get(nameIndex).toLowerCase().trim();
+        } else {
+            System.err
+                    .println("No" + flag + " argument is specified." + " Please use the format: " + flag + " flagName");
+            System.exit(0);
+        }
+        return flagName;
+    }
+
     /**
      * @param args
      */
     public static void main(String[] args) {
         List<String> argsList = new ArrayList<String>(Arrays.asList(args));
-        System.out.println(argsList);
+
+        String techniqueName = getArgName(argsList, "-technique");
+
+        String directoryName = getArgName(argsList, "-directory");
+
+        String coverageName = getArgName(argsList, "-coverage");
+
+        String orderName = getArgName(argsList, "-order");
+
+        String projectName = getArgName(argsList, "-project");
+
+        String testType = getArgName(argsList, "-testType");
```

Comment:
If you directly get the String this way and start using it then the user will not be able to specify more than one configuration and as a result this class will not be able to directly generate figures 17, 18, and 19. You can either check for a delimiter when you get the configurations or you can directly just get the configurations from the files.

From doing the former option to generate figure 17 you would have to specify something like "-projects crystal,jfreechart,jodatime,synoptic,xml_security". If you directly parse the files from the directory, whatever projects it sees, it will generate a row/column for in the table.